Assets/Scripts/Player/Mothership.cs

```diff
@@ -0,0 +1,101 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using ManyTools.Variables;
+
+public class Mothership : MonoBehaviour
+{
+    #region Private Fields
+    [SerializeField]
+    private FloatReference _life;
+    [SerializeField]
+    private FloatReference _speed;
+    [SerializeField]
+    private FloatReference _bulletForce;
+    private int _cyanShips;
+    #endregion
+    #region Public Fields
+    public GameObject bulletPrefab;
+    public Transform bulletSpawn;
+    #endregion
+    #region Unity Callback
+    void Start()
+    {
+        
+    }
+
+    void Update()
+    {
+        MothershipMovement();
+        MothershipShoot();
+        MothershipCyanShoot();
+    }
+    #endregion
+    #region Commands Mothership
+    /// <summary>
+    /// WASD move to up, down, left and right; Q and E rotate the ship
+    /// </summary>
+    private void MothershipMovement()
+    {
+        float _timeVelocity = _speed * Time.deltaTime;
+        if (Input.GetKey(KeyCode.W) && transform.position.y < 14)
+        {
+            transform.position += (Vector3.up * _timeVelocity);
+        }
+
+        if (Input.GetKey(KeyCode.S) && transform.position.y > -14)
+        {
+            transform.position += (Vector3.down * _timeVelocity);
+        }
+
+        if (Input.GetKey(KeyCode.A) && transform.position.x > -23)
+        {
+            transform.position += (Vector3.left * _timeVelocity);
+        }
+
+        if (Input.GetKey(KeyCode.D) && transform.position.y < 23)
+        {
+            transform.position += (Vector3.right * _timeVelocity);
+        }

```

Comment:
- Use o ``transform.Translate`` ao invés de setar o ``transform.position`` repetidamente. O transform.position tem um custo superior.
- Imagino que seja placeholder, mas os limites precisos de posição não devem ser hardcoded no código. Eles não devem sequer aparecer no código.
- Faça mude o movimento para aceitar um Vector2
```suggestion
        Vector3 movement = new Vector3(Input.GetAxis("Horizontal"), Input.GetAxis("Vertical"))
        transform.Translate(movement * timeVelocity);
```